basiclti/basiclti-impl/src/test/org/sakaiproject/util/foorm/FoormSakaiTest.java

```diff
@@ -0,0 +1,242 @@
+ *
+ * Copyright (c) 2011 The Sakai Foundation
+ *
+ * Licensed under the Educational Community License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.opensource.org/licenses/ECL-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **********************************************************************************/
+
+package org.sakaiproject.util.foorm;
+
+import static org.junit.Assert.*;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertFalse;
+
+import org.junit.Test;
+
+import java.lang.Class;
+import java.math.BigDecimal;
+
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Timestamp;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Date;
+import java.util.Properties;
+import java.util.TimeZone;
+import java.util.Arrays;
+
+import java.text.SimpleDateFormat;
+import java.time.Clock;
+import java.time.Instant;
+import java.time.LocalDateTime;
+import java.time.ZoneOffset;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeFormatterBuilder;
+import java.time.temporal.TemporalAccessor;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.sakaiproject.lti.api.LTIService;
+
+/*
+ * This is originally based on the FoormTest.java file but modified to use JUnit
+ * code and conventions. It is targeted for a change to address the issue tested
+ * testIntegerField.  Most code that did not address the parsing issue has been deleted.
+ *
+ * Sample code for testing hsql and oracle databases has been kept.  It is not yet
+ * suitable for actual junit testing since it creates direct output rather than
+ * allowing testing assertions but it seems a shame to have to recreate the functionality
+ * later from scratch.
+ *
+ * There will need to be some code refactoring to make it possible to
+ * fully unit test the LTI code.
+ *
+ */
+@Slf4j
+public class FoormSakaiTest {
+	String[] vendors = {"mysql", "oracle", "hsqldb"};
+
+	/* This is a weird unit test.  It's goal is to generate the DDL and optionall dump it. 
+	 * when a Shell Environment variable is set.
+	 *
+	 * export SAKAI_FOORM_DDL_DUMP=yes
+	 * mvn test
+	 *
+	 */
+	@Test
+	public void testCreateDDL() {
+		boolean doDump = "yes".equals(System.getenv("SAKAI_FOORM_DDL_DUMP"));
+		String [] sqls;
+	    Foorm foorm = new Foorm();
```

Comment:
It seems there are spacing issues with "Foorm foorm" and "sqls" below.

Reply:
Thanks.  Spacing is the final frontier.